Analysis/Tasks/PWGHF/HFMCMatchingValidation.cxx

```diff
@@ -0,0 +1,200 @@
+     {"hMCEvt_CouterPerCollision_LambdaC", "Event counter - LambdaC; entries", {HistType::kTH1F, {{10, -0.5, +9.5}}}}}};
+
+  void process(aod::McCollision const& mccollision, aod::McParticles const& particlesMC)
+  {
+    int cpercollision = 0;
+    int cbarpercollision = 0;
+    int bpercollision = 0;
+    int bbarpercollision = 0;
+    double sumPxDau, sumPyDau, sumPzDau, momentum_check;
+    double pxDiff, pyDiff, pzDiff;
+
+    //Particles and relative decay checked in the second part of the task  D+->piKpi, D*+->piKpi, D0->K\pi, LambdaC->\pK\pi
+    std::array<int, 4> PDGArrayParticle = {411, 413, 421, 4122};
+    std::array<std::array<int, 3>, 4> arrPDGFinal = {{{211, 211, -321}, {211, 211, -321}, {321, -211, 0}, {2212, -321, 211}}};
+    int counter[4] = {0, 0, 0, 0};
+    std::vector<int>* listDaughters = new std::vector<int>;
+
+    for (auto& particle1 : particlesMC) {
+      //counting HF quarks
+      int particlePdgCode = particle1.pdgCode();
+      int motherPdgCode = particlesMC.iteratorAt(particle1.mother0()).pdgCode();
+      if (particlePdgCode != motherPdgCode) {
+        switch (particlePdgCode) {
+          case 4:
+            cpercollision++;
+            break;
+          case -4:
+            cbarpercollision++;
+            break;
+          case 5:
+            bpercollision++;
+            break;
+          case -5:
+            bbarpercollision++;
+            break;
+        }
+      }
+
+      sumPxDau = 0;
+      sumPyDau = 0;
+      sumPzDau = 0;
+      momentum_check = 1;
+      listDaughters->clear();
+
+      // Checking the decay of the particles and the momentum conservation
+      for (int iD = 0; iD < PDGArrayParticle.size(); iD++) {
+        if (particlePdgCode == PDGArrayParticle[iD]) {
+          //Check whether the particle is the expected one
+          if (RecoDecay::isMatchedMCGen(particlesMC, particle1, PDGArrayParticle[iD], true)) {
+            RecoDecay::getDaughters(particlesMC, particle1.globalIndex(), listDaughters, arrPDGFinal[iD], -1);
+            if (listDaughters->size() <= arrPDGFinal[iD].size()) {
+              counter[iD]++;
+            }
+            for (int i = 0; i < listDaughters->size(); i++) {
+              if (listDaughters->at(i) < 0)
+                continue;
+              auto daughter = particlesMC.iteratorAt(listDaughters->at(i));
+              sumPxDau += daughter.px();
+              sumPyDau += daughter.py();
+              sumPzDau += daughter.pz();
+            }
+            pxDiff = particle1.px() - sumPxDau;
+            pyDiff = particle1.py() - sumPyDau;
+            pzDiff = particle1.pz() - sumPzDau;
+            if (std::abs(pxDiff) > 0.001 || std::abs(pyDiff) > 0.001 || std::abs(pzDiff) > 0.001)
+              momentum_check = 0;
+            double p_magnitude_diff = RecoDecay::sqrtSumOfSquares(pxDiff, pyDiff, pzDiff);
+            double pt_magnitude_diff = RecoDecay::sqrtSumOfSquares(pxDiff, pyDiff);
```

Comment:
Not strictly necessary but for simplicity, you can call directly `RecoDecay::P` and `RecoDecay::Pt`.